### PR TITLE
LPS-68775 Cannot add an upgraded organization as parent of newly added organizations

### PR DESCRIPTION
**Pull request by pei-jung**
/cc @mdelapenya @drewbrokke

### CODE REVIEW

**Comment by pei-jung** [comment]
ci:reopen

**Comment by pei-jung** [comment]
ci:retest

**Comment by JorgeFerrer** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: portal-impl/src/com/liferay/portal/upgrade/UpgradeProcess_7_0_3.java [hunk 1]
```diff
@@ -36,6 +37,7 @@ public int getThreshold() {
 	protected void doUpgrade() throws Exception {
 		upgrade(UpgradeMessageBoards.class);
 		upgrade(UpgradeModules.class);
+		upgrade(UpgradeOrganization.class);
```
Comment (mdelapenya) [inline portal-impl/src/com/liferay/portal/upgrade/UpgradeProcess_7_0_3.java]: Hey @JorgeFerrer, I've just realised this line is not properly ordered. It should be after UpgradeOracle

Sorry!

File: portal-impl/src/com/liferay/portal/upgrade/v7_0_3/UpgradeOrganization.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+public class UpgradeOrganization extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		updateRegularOrganizations();
+	}
+
+	protected void updateRegularOrganizations() throws Exception {
+		try (LoggingTimer loggingTimer = new LoggingTimer()) {
+			runSQL("update Organization_ set type_=\"organization\"");
```
Comment (JorgeFerrer) [inline portal-impl/src/com/liferay/portal/upgrade/v7_0_3/UpgradeOrganization.java]: Hey @mdelapenya @pei-jung,

So what happens with this line to all of our customers who are using organization types for their own purposes?

For example, an education portal may be using organization types such as "School Group", "School", "Degree", "Class", ... and suddenly they all become "organization". I can imagine them being very unhappy about such a change being done massively to their database.

**Comment by JorgeFerrer** [comment]
Hey guys,

I'm not sure if this is the right approach to fix this issue. If we want to do it with an upgrade then I think we need to check the configuration and see if "regular-organization" is no longer there (the fact that it's not there by default doesn't mean that it cannot be added, even if it's unlikely). Then, the SQL should only change organizations with type "regular-organization" to "organization" and not all organizations.

An alternative (which I think is what we were doing and which I think we should be doing anyway) is to make "organization" the default organization type. That means that if any Organization has an organization type which is not recognized, we log it but then behave as if it was "organization". This is akin to what we do with themes. If any site has in the db a theme that does not exist, we just assume it's "classic".

Makes sense?

Jorge

**Comment by pei-jung** [comment]
Hey @JorgeFerrer,

I think you might have missed it. I added a commit to only update the organization type to `organizaiton` if the type is not specified in the portal property da04cff.

**Comment by JorgeFerrer** [comment]
You are right, I had missed that commit. I've now looked at it and it looks good.

Having said that, I also think that we should make the code that deals with types more robust so that if it finds a type that it doesn't know about it treats it as organization (or maybe, whatever is the first existing type). That will help fix the bug even when the upgrade has not been run.

**Comment by pei-jung** [comment]
That makes sense. I'll update the code to set `organization` as the default for unrecognized organization types.

Closing the PR for additional change. Will resend.

**Comment by pei-jung** [comment]
Resent as https://github.com/JorgeFerrer/liferay-portal/pull/801. No changes made.

**Pull request closed without merging**